drivers/wifi/siwx91x/siwx91x_wifi.c

```diff
@@ -173,6 +173,63 @@ static int siwx91x_ap_disable(const struct device *dev)
 	return ret;
 }
 
+static int siwx91x_ap_sta_disconnect(const struct device *dev, const uint8_t *mac_addr)
+{
+	ARG_UNUSED(dev);
+	sl_wifi_interface_t interface;
+	sl_mac_address_t mac = { };
+	int ret;
+
+	__ASSERT(mac_addr, "mac_addr cannot be NULL");
+
+	memcpy(mac.octet, mac_addr, ARRAY_SIZE(mac.octet));
+	interface = sl_wifi_get_default_interface();
+
+	ret = sl_wifi_disconnect_ap_client(interface, &mac, SL_WIFI_DEAUTH);
```

Comment:
Just curiosity: I would naturally use `sl_wifi_disconnect_ap_client(SL_WIFI_AP_INTERFACE | SL_WIFI_2_4GHZ_INTERFACE, &mac, SL_WIFI_DEAUTH);`. Why it wouldn't work?